app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,25 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:success] = 'Post created!'
+      redirect_to "/users/#{current_user.id}/posts"
```

Comment:
It would be simpler if you would redirect just to the post 
```
 redirect_to @post
```

Reply:
- [ ] Because of these validations:
```
  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
on your post model I was unable to create posts, because when initialized the likes and comments counter is nil, you can make it by default 0